pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/bestpractices/xml/UnusedPrivateField.xml

```diff
@@ -540,6 +540,17 @@ public class Foo {
         ]]></code>
     </test-code>
 
+    <test-code>
+        <description>#907 UnusedPrivateField false-positive with @FXML</description>
```

Comment:
A test case for `@Deprecated` would be nice too.

Note that if you test the type of the Annotation instead of its name, as I describe above, you'll need to add the import statement for the `@FXML` test case.

Reply:
Actually,` @Deprecated` should be added to rule `UnusedPrivateMethod` and I will do it in next commit:)

Reply:
I'd not change `UnusedPrivateMethod` to not report on `@Deprecated`. If it's private, removing it won't break the contract / API, meaning it can be safely removed. On hindsight, the same applies to this very rule...

Reply:
Yes, the same applies to this rule. The use case is if another class or framework accesses the field or method reflectively, though I agree that's rare enough (JavaFX is the example we have for fields). Maybe you're right, and we should wait until someone needs the feature implemented

Reply:
Yeah, that should be uncommon enough to justify a suppression. Non the less, your suggestion of using `TypeMultiProperty.named("ignoredAnnotations")` seems absolutely right.

Reply:
@jsotuyod I thought I could do that myself when I merge if @djydewang doesn't. I'll merge tonight if I don't hear from any of you

Reply:
I'd love to carry it out but I need some time to get familiar with the code:) It would be great also if you could give me a good demostration in this PR.

Reply:
@djydewang Well this change amounts to defining a static final property descriptor (documentation [here](https://pmd.github.io/pmd-6.1.0/pmd_userdocs_extending_defining_properties.html)), then defining it in the rule's constructor, and using `getProperty` instead of accessing your static final set of strings. Btw I think, since we now use a set of strings and not Class, the property should be a `StringMultiProperty`

The point of that is to let users configure the names of the annotations they want to ignore themselves, like described [here](https://pmd.github.io/pmd-6.1.0/pmd_userdocs_using_properties.html). But we can still put FXML and Deprecated as default values.

If you feel like doing it yourself, just tell me and I'll wait for your next update. Then I can review and help you if you need it ;) Otherwise I can do it myself and you can look at the updated code.

Reply:
Cool, good night then !

Reply:
Thank you for your detailed guidance. It's really a friendly community:)
Done.